staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go

```diff
@@ -268,15 +270,16 @@ func (gvs GroupVersions) KindForGroupVersionKinds(kinds []GroupVersionKind) (Gro
 
 // bestMatch tries to pick best matching GroupVersionKind and falls back to the first
 // found if no exact match exists.
-func bestMatch(kinds []GroupVersionKind, targets []GroupVersionKind) GroupVersionKind {
-	for _, gvk := range targets {
-		for _, k := range kinds {
-			if k == gvk {
-				return k
-			}
+func bestMatch(kinds []GroupVersionKind, targets map[GroupVersionKind]struct{}) GroupVersionKind {
+	for _, k := range kinds {
+		if _, ok := targets[k]; ok {
+			return k
 		}
 	}
-	return targets[0]
```

Comment:
This previously honored the order of gvs. As a map, it no longer does.

Reply:
Thanks for the reminder - which doesn't seem to be recorded in existing comment in code.

What if we produce a map[GroupVersionKind]struct{} for kinds ?
that way the order is preserved for targets.